frugalos_config/src/cancelable_future.rs

```diff
@@ -0,0 +1,94 @@
+use futures::{Async, Future, Poll};
+use std::sync::mpsc;
+use Error;
+
+/// Signalを受信するための構造体であり、Futureとしても扱うことができる。  
+/// `make_signal`関数によって、対応するSenderとの組として生成される。
+pub struct SignalReceiver(mpsc::Receiver<()>);
```

Comment:
This struct does only `cancel` operation but I feel the name is too abstract.

Reply:
You're right; at least, if we would adopt this name, this struct should serve some kinds of operations to receive signal.